src/DataTypes/Serializations/SerializationString.cpp

```diff
@@ -334,9 +334,12 @@ bool SerializationString::tryDeserializeTextEscaped(IColumn & column, ReadBuffer
     return read<bool>(column, [&](ColumnString::Chars & data) { readEscapedStringInto(data, istr); return true; });
 }
 
-void SerializationString::serializeTextQuoted(const IColumn & column, size_t row_num, WriteBuffer & ostr, const FormatSettings &) const
+void SerializationString::serializeTextQuoted(const IColumn & column, size_t row_num, WriteBuffer & ostr, const FormatSettings & settings) const
 {
-    writeQuotedString(assert_cast<const ColumnString &>(column).getDataAt(row_num), ostr);
+    if (!settings.values.escape_quote_with_quote)
+        writeQuotedString(assert_cast<const ColumnString &>(column).getDataAt(row_num), ostr);
+    else
+        writeQuotedStringPostgreSQL(assert_cast<const ColumnString &>(column).getDataAt(row_num).toView(), ostr);
```

Comment:
it will be a bit more readable if it was `if (setting) ..  else ..` instead of `if (!setting) ... else ...`